cpp/src/arrow/array/util.h

```diff
@@ -56,6 +56,14 @@ Result<std::shared_ptr<Array>> MakeArrayFromScalar(
 
 namespace internal {
 
+/// \brief Swap endian of each element in a generic ArrayData
+/// \param[in] data the array contents
+/// \param[in] type the array type
+/// \return the resulting Array instance whose elements were swapped
+ARROW_EXPORT
+Result<std::shared_ptr<ArrayData>> SwapEndianArrayData(
+    std::shared_ptr<ArrayData>& data, const std::shared_ptr<DataType>& type);
+
 /// Given a number of ArrayVectors, treat each ArrayVector as the
```

Comment:
Hmm... did I miss the tests for `SwapEndianArrayData` or didn't you add any?
I would expect some simple basic tests, at least to exercise nulls, numbers, structs, lists, decimals...

Reply:
Thank you for your comments. I will address it.

Regarding the test, there are only integration tests as discussed at https://github.com/apache/arrow/pull/7507#issuecomment-657310102. I will add tests for `SwapEndianArrayData`.